From 907ded5a1d81c443d38e64bafc080ce32abcdef9 Mon Sep 17 00:00:00 2001
From: Daniel Stenberg <daniel@haxx.se>
Date: Sun, 18 Jul 2010 19:47:21 +0200
Subject: [PATCH] libssh2_wait_socket: reset error code to "leak" EAGAIN less

[upstream commit 697b4e8ed7268e635ae7eaaf6b77b522e068467f]

Since libssh2 often sets LIBSSH2_ERROR_EAGAIN internally before
_libssh2_wait_socket is called, we can decrease some amount of
confusion in user programs by resetting the error code in this function
to reduce the risk of EAGAIN being stored as error when a blocking
function returns.

Signed-off-by: Kamil Dudka <kdudka@redhat.com>
---
 src/session.c |    6 ++++++
 1 files changed, 6 insertions(+), 0 deletions(-)

diff --git a/src/session.c b/src/session.c
index aefc86f..ff6e46d 100644
--- a/src/session.c
+++ b/src/session.c
@@ -509,6 +509,12 @@ int _libssh2_wait_socket(LIBSSH2_SESSION *session)
     int dir;
     int rc;
 
+    /* since libssh2 often sets EAGAIN internally before this function is
+       called, we can decrease some amount of confusion in user programs by
+       resetting the error code in this function to reduce the risk of EAGAIN
+       being stored as error when a blocking function has returned */
+    session->err_code = LIBSSH2_ERROR_NONE;
+
     FD_ZERO(&fd);
     FD_SET(session->socket_fd, &fd);
 
-- 
1.7.1

